aiven_mysql_migrate/dump_tools.py

```diff
@@ -117,3 +117,232 @@ def get_import_command(self) -> List[str]:
 
         return cmd
+
+
+class MyDumperTool(MySQLMigrationToolBase):
+    """MyDumper tool using mydumper/myloader."""
+    def __init__(
+        self,
+        source: MySQLConnectionInfo,
+        target: MySQLConnectionInfo,
+        databases: List[str],
+        skip_column_stats: bool,
+        *,
+        dump_tool_name: MySQLMigrateTool = MySQLMigrateTool.mydumper,
+    ):
+        super().__init__(source, target, databases, skip_column_stats, dump_tool_name=dump_tool_name)
+        self.temp_dir: Optional[tempfile.TemporaryDirectory] = None
+        self.temp_cnf_file: Optional[Path] = None
+        self.temp_target_cnf_file: Optional[Path] = None
+        self.dump_output_dir: Optional[Path] = None
+
+    def setup(self) -> None:
+        """Setup temporary resources for migration."""
+
+        if not self.temp_dir:
+            self.temp_dir = self._create_temp_directory()
+
+        if not self.temp_cnf_file:
+            self.temp_cnf_file = self._create_temp_cnf_file(self.source, "source.cnf")
+
+        if not self.temp_target_cnf_file:
+            self.temp_target_cnf_file = self._create_temp_cnf_file(self.target, "target.cnf")
+
+        if not self.dump_output_dir:
+            self.dump_output_dir = self._get_dump_output_dir()
+
+    def execute_migration(self, migration_method: MySQLMigrateMethod) -> Optional[str]:
+        """
+        Execute the migration and extract GTID from metadata file if available.
+
+        Args:
+            migration_method: The migration method (dump or replication)
+
+        Returns:
+            GTID string for replication setup, or None
+        """
+        self.setup()
+        dump_cmd = self.get_dump_command(migration_method)
+        import_cmd = self.get_import_command(migration_method)
+        # LOGGER.debug("dump command: %s", " ".join(shlex.quote(arg) for arg in dump_cmd))
+        # LOGGER.debug("import command: %s", " ".join(shlex.quote(arg) for arg in import_cmd))
+        self._gtid = self.process_executor.execute_piped_commands(
+            dump_cmd=dump_cmd,
+            import_cmd=import_cmd,
+            target=self.target,
+            dump_tool=self.dump_tool_name,
+            dump_output_dir=self.dump_output_dir,
+            backup_dir=Path(self.temp_dir.name) if self.temp_dir else None
+        )
+
+        # If we need GTID for replication, extract it from metadata file
+        if migration_method == MySQLMigrateMethod.replication:
+            gtid = self._extract_gtid_from_metadata()
+            if gtid:
+                self._gtid = gtid
+            else:
+                raise ReplicaSetupException("Failed to extract GTID from mydumper metadata for replication setup")
+
+        return self._gtid
+
+    def _create_temp_cnf_file(self, connection_info: MySQLConnectionInfo, filename: str = "connection.cnf") -> Path:
+        """Create temporary .cnf file with credentials for secure password handling."""
+        if not self.temp_dir:
```

Comment:
This `if`s to check the dirs/files are not needed, as we call setup, if you want to make `mypy` happy, it could be an assert.